ykrt/src/deopt.rs

```diff
@@ -113,16 +114,17 @@ extern "C" fn __llvm_deoptimize(
             "push rcx",
             "push rdx",
             "push rax",
+            "mov r10, r8", // Store guardid in r10 so we can use r8 for the arguments.
```

Comment:
Just to confirm: since `r10` is a caller-saved register we can freely use it here to temporarily store one of the arguments.

Reply:
Yeah, I checked that. Why is my nose growing?